src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs

```diff
@@ -267,45 +289,42 @@ void UpdateSemanticProperties(IView view)
 		/// <inheritdoc/>
 		protected override WSize ArrangeOverride(WSize finalSize)
 		{
-			if (_renderer == null)
-			{
-				return base.ArrangeOverride(finalSize);
-			}
-
-			var width = ItemWidth == default ? finalSize.Width : ItemWidth;
-			var height = ItemHeight == default ? finalSize.Height : ItemHeight;
-			var size = new WSize(width, height);
-
-			return base.ArrangeOverride(size);
+			return base.ArrangeOverride(finalSize);
 		}
 
 		/// <inheritdoc/>
 		protected override WSize MeasureOverride(WSize availableSize)
 		{
-			if (_renderer == null)
+			if (_renderer is null)
 			{
 				return base.MeasureOverride(availableSize);
 			}
 
 			var width = ItemWidth == default ? availableSize.Width : ItemWidth;
 			var height = ItemHeight == default ? availableSize.Height : ItemHeight;
-			var measuredSize = base.MeasureOverride(new WSize(width, height));
-			var finalWidth = Max(measuredSize.Width, ItemWidth);
-			var finalHeight = Max(measuredSize.Height, ItemHeight);
-			var finalSize = new WSize(finalWidth, finalHeight);
-			var frameworkElement = Content as FrameworkElement;
-			var visualElement = _renderer.VirtualView as VisualElement;
-			var margin = _renderer.VirtualView.Margin;
-
-			frameworkElement.Margin = WinUIHelpers.CreateThickness(margin.Left, margin.Top, margin.Right, margin.Bottom);
-			visualElement.Layout(new Rect(0, 0, finalWidth, finalHeight));
-
-			if (CanMeasureContent(frameworkElement))
+			var measureSize = base.MeasureOverride(new WSize(width, height));
+
+			width = measureSize.Width;
+			height = measureSize.Height;
+
+			if (ItemHeight == default && ItemWidth == default)
+			{
+				if(double.IsFinite(availableSize.Width))
```

Comment:
I'm not too sure if the code here is correct. 

I pulled it from the older code that was on this PR
https://github.com/dotnet/maui/commit/862fa07227bdad4a26175d785492f697eb37ab5e#

Reply:
it looks like this was added for grouping reasons
https://github.com/xamarin/Xamarin.Forms/blame/88b70ad2824a4f40fa2ef056439ada9dc3e85ab4/Xamarin.Forms.Platform.UAP/CollectionView/ItemContentControl.cs#L222-L226

So we should test grouping